openquake/hazardlib/geo/surface/base.py

```diff
@@ -229,8 +236,14 @@ def get_rx_distance(self, mesh):
         top_edge = self.mesh[0:1]
         dists = []
         if top_edge.lons.shape[1] < 3:
-
             i = 0
+
+            if ((self.__class__.__name__ == 'KiteSurface') and
+                (numpy.isnan(top_edge.lons[0, i]) or
```

Comment:
Why not the more idiomatic `isinstance(self, KiteSurface)`?

Reply:
ditto